cpp/src/io/comp/nvcomp_adapter.cpp

```diff
@@ -140,13 +141,11 @@ size_t batched_decompress_temp_size(compression_type compression,
 {
   size_t temp_size         = 0;
   auto const nvcomp_status = [&]() {
-    try {
-      return batched_decompress_get_temp_size_ex(
-        compression, num_chunks, max_uncomp_chunk_size, &temp_size, max_total_uncomp_size);
-    } catch (cudf::logic_error const& err) {
-      return batched_decompress_get_temp_size(
-        compression, num_chunks, max_uncomp_chunk_size, &temp_size);
-    }
+    auto maybe_result = batched_decompress_get_temp_size_ex(
+      compression, num_chunks, max_uncomp_chunk_size, &temp_size, max_total_uncomp_size);
+    if (maybe_result.has_value()) { return *maybe_result; }
+    return batched_decompress_get_temp_size(
+      compression, num_chunks, max_uncomp_chunk_size, &temp_size);
```

Comment:
This is better in theory (no branch)
```suggestion
return maybe_result.value_or(
  batched_decompress_get_temp_size(compression, num_chunks, max_uncomp_chunk_size, &temp_size));
```

Going one step further, the whole lambda could be replaced with
```
auto const nvcomp_status = batched_decompress_get_temp_size_ex(
         compression, num_chunks, max_uncomp_chunk_size, &temp_size, max_total_uncomp_size)
  .value_or(
    batched_decompress_get_temp_size(compression, num_chunks, max_uncomp_chunk_size, &temp_size));
```

Reply:
yes, that looks much nicer, thanks!